.github/workflows/cargo-check.yaml

```diff
@@ -0,0 +1,33 @@
+name: Cargo Check
+on:
+  push:
+    branches:
+    - '**'
+    tags-ignore:
+    - v[0-9]+.[0-9]+.[0-9]+*
```

Comment:
The benefit of running this always is that we check for `try-runtime` and `runtime-benchmarks` builds.
Otherwise, a code that breaks them could become a part of release.

Should we still run this as part of release then?
Maybe we can bunch it up together with `test` so they can reuse each other's cache?